pkg/query-service/app/traces/v3/query_builder.go

```diff
@@ -501,26 +501,35 @@ func addOffsetToQuery(query string, offset uint64) string {
 // step is in seconds
 func PrepareTracesQuery(start, end int64, panelType v3.PanelType, mq *v3.BuilderQuery, options v3.QBOptions) (string, error) {
 	// adjust the start and end time to the step interval
-	start = start - (start % (mq.StepInterval * 1000))
-	end = end - (end % (mq.StepInterval * 1000))
+	var queryStart, queryEnd int64
+
+	if panelType == v3.PanelTypeValue || panelType == v3.PanelTypeList || panelType == v3.PanelTypeTable {
+		queryStart = start
+		queryEnd = end
+	} else {
+		// adjust the start and end time to the step interval for other panel types
+		queryStart = start - (start % (mq.StepInterval * 1000))
+		queryEnd = end - (end % (mq.StepInterval * 1000))
+	}
```

Comment:
I take back my previous comment. It's just cleaner to do something like this.
```go
if panelType == v3.PanelTypeGraph {
	// adjust the start and end time to the step interval for other panel types
	start = start - (start % (mq.StepInterval * 1000))
	end = end - (end % (mq.StepInterval * 1000))
}
```

Reply:
I'll make the changes and revert back to you

Reply:
@srikanthccv I have made the changes, can you review?